ray_lightning/launchers/ray_launcher.py

```diff
@@ -298,16 +298,20 @@ def _wrapping_function(
         trainer.strategy.local_rank = self._strategy.local_rank
         set_cuda_device_if_used(trainer.strategy)
 
+        # Set operations to deterministic in this worker when required
+        if trainer._accelerator_connector.deterministic:
+            trainer._accelerator_connector._init_deterministic(True)
+
         results = function(*args, **kwargs)
 
         if trainer is not None:
-            return self._collect_rank_zero_results(trainer, results)
-        else:
-            return None
+            results = self._collect_rank_zero_results(trainer, results)
+
+        if results is None:
+            trainer._teardown()
```

Comment:
Why do we need to tear down `trainer` only when `local_rank` or `global_rank` is `!=0`?